Compiling with clang on Linux, libc++ vs libstdc++

When I compile Pangolin on Linux with clang, it compiles with `libc++`: https://github.com/stevenlovegrove/Pangolin/blob/4335035057dc26f98b1a498b9b4843015fc9e5c5/CMakeLists.txt#L53

This causes issue when later using it in another project which is compiled with the default `libstdc++`. You get errors like:

```
undefined reference to `pangolin::CreateWindowAndBind(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, int, int, pangolin::Params const&)'
```

If that particular line in the `CMakeLists.txt` is removed, clang defaults to `libstdc++` on Linux and everything works fine.

Do you remember why that line is there? It looks like it has been for 5 or more years. Maybe back then things were different, but I believe that clang defaults to the native STL implementation on each platform, i.e. to `libstdc++` on Linux and `libc++` on macOS. So maybe that line can be removed?

--- a/.travis.yml
+++ b/.travis.yml
@@ -8,6 +8,14 @@ before_install:
   - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then brew update          ; fi
   - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then brew install eigen glew ; fi
 
+addons:
+  apt:
+    sources:
+      - ubuntu-toolchain-r-test
+    packages:
+      # install newer libstdc++ (from ubuntu-toolchain-r/test PPA), since otherwise clang fails with old libstdc++-4.8
+      - libstdc++-5-dev
+
 language: cpp
 
 compiler:

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -49,9 +49,6 @@ endif()
 
 if(NOT MSVC)
     set( CMAKE_CXX_FLAGS "-std=c++0x -Wall -Wextra ${CMAKE_CXX_FLAGS}" )
-    if(_CLANG_)
-        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -stdlib=libc++")
-    endif()
 endif()
 
 if(ANDROID)

